lambdas/src/common/config/config-service.ts

```diff
@@ -34,23 +35,23 @@ export class ConfigService {
         if (ssmParameters.length === 0) {
             throw new Error(`No client config found. Invalid client id encountered: ${clientId}`);
         }
-        await this.getParametersByAbsolutePath(ssmParameters, clientId);
+        await this.setParametersByAbsolutePath(ssmParameters, clientId);
     }
 
     public async initConfigWithCriIdentifierInPath(clientId: string, parameterPrefix: string, paramNameSuffix: string) {
-        const ssmParameters = await this.getParameters([
-            `/${AWS_STACK_NAME_PREFIX}/${parameterPrefix}/${paramNameSuffix}`,
-        ]);
+        const ssmParameters = await this.getCriIdentifierParameters([`/${parameterPrefix}/${paramNameSuffix}`]);
         if (ssmParameters.length === 0) {
-            throw new Error(`Invalid parameter beginning with ${parameterPrefix} encountered`);
+            logger.info(`Invalid parameter beginning with ${parameterPrefix} encountered`);
         }
-        await this.getParametersByAbsolutePath(ssmParameters, clientId);
+        await this.setParametersByAbsolutePath(ssmParameters, clientId);
     }
 
-    private async getParametersByAbsolutePath(ssmParameters: Parameter[], identifier: string) {
+    private async setParametersByAbsolutePath(ssmParameters: Parameter[], identifier: string) {
         const clientConfigEntries: Map<string, string> =
             this.clientConfigurations.get(identifier) || new Map<string, string>();
 
+        if (ssmParameters.length === 0) return;
```

Comment:
Sorry I'm not sure I'm following this so -
1. Line 35 we are checking if we have no ssm params and throwing an error if that's true
2. Line 38 we must have ssmParameters but then we are setting the parameters by absolute path which seems to actually set the clientConfigurations?
3. Line 53 we then check again to see if we have ssm params?

Reply:
Hmm, I have moved the return to be specifically in the initConfigWithCriIdentifierInPath function and removed the additional length check from line 53. Agreed this class is slightly confusing still though, are you happy with just this change or does it need refactoring further in this PR?

Reply:
Thanks for removing the additional check - happy for any additional refactoring to fall outside of the scope of this PR.

What is your thinking for changing the `throw` to a `log` + `return`?  Would that mean our `initialiseClientConfigMiddleware` can still continue even if we know we're not going to be able to validate our evidence requested in the `validateJwtMiddleware`?

Reply:
The evidence-requested ssm param is optional, as we do not want to break other CRIs by enforcing it to be there.

This previously threw an error because we added the strengthScore ssm param to all CRIs in the CF template. Now we have removed that and used terraform to deploy evidence-requested only to the CRI's we want. 

Currently evidence-properties is the only potential parameter passed into initConfigWithCriIdentifierInPath.

Reply:
The alternative was to set the evidence-properties ssm param in the CF template for all CRI's and not to use terraform to deploy it. Because we have used terraform we need to handle it not being there.

Reply:
Ah ok and the checks in `validateJwtMiddleware` now check if we have a `configuredStrengthScore` _and_ if the `evidenceRequested` matches that. Ok cool that makes sense 👍